File: iOS/DataPool/Sources/DataPool/Domain/PoolMemberRepository.swift
import Core

public protocol PoolMemberRepository {
    func getPoolMembers(
        poolId: String,
        next: String?
    ) async -> Result<CursorPage<PoolMember>, Error>

    func removeGambler(
        poolId: String,
        gamblerId: String
    ) async -> Result<Void, Error>
}

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/domain/PoolMemberRepository.kt
package com.felipearpa.tyche.data.pool.domain

import com.felipearpa.tyche.core.paging.CursorPage

internal interface PoolMemberRepository {
    suspend fun getPoolMembers(
        poolId: String,
        next: String? = null,
    ): Result<CursorPage<PoolMember>>

    suspend fun removeGambler(poolId: String, gamblerId: String): Result<Unit>
}
